unicode/graphemes/Graphemes.g4

```diff
@@ -12,7 +12,7 @@ fragment TextPresentationSequence: EmojiPresentationCharacter VS15;
 fragment EmojiPresentationSequence: TextPresentationCharacter VS16;
 fragment EmojiCharacter: [\p{Emoji}];
 fragment EmojiModifierSequence:
-    [\p{Grapheme_Cluster_Break=E_Base}\p{Grapheme_Cluster_Break=E_Base_GAZ}] [\p{Grapheme_Cluster_Break=E_Modifier}];
+    . [\p{Grapheme_Cluster_Break=E_Modifier}];
```

Comment:
What's the need for the dot here? I'd just removed not supported grapheme.

Reply:
https://www.unicode.org/reports/tr29/#Default_Grapheme_Cluster_Table

Seems E_Base, E_Modifier, E_Base_GAZ are all obsoleted.

Reply:
Yes, I know, I asked about the dot, not about these graphemes.

Reply:
Removing a specific character set is not the same as allowing a single character of any value. The rule accepted two characters before, it still does. The change you propose says that two characters are not acceptable, that the fragment denotes now one character. Are you sure you are correct? It really doesn't not sound correct. `[\p{Grapheme_Cluster_Break=E_Base}\p{Grapheme_Cluster_Break=E_Base_GAZ}]` != ''.

Reply:
I don't understand this grammar, or the other in `unicode/`. I'm reading `unicode/readme.md` and I can't understand what is the purpose. Is it just to test Antlr4 unicode expressions in a lexer set? It seems it's a test suite of some kind, but it doesn't say so *anywhere* in the readme. If it is, it should say so.

The change to get it to work for 4.9.3 shouldn't even be here since it has nothing to do with a build. If someone wants to modify this, by my guest. I'll roll whatever changes into the PR.

Reply:
The rule `EmojiModifierSequence` was removed entirely from the similar grammar from the official repository: https://github.com/antlr/antlr4/pull/3323/files I suggest doing the same.

> I don't understand this grammar, or the other in unicode/. I'm reading unicode/readme.md and I can't understand what is the purpose. Is it just to test Antlr4 unicode expressions in a lexer set? It seems it's a test suite of some kind, but it doesn't say so anywhere in the readme.

It seems like so.

Reply:
According to [Unicode.org](https://unicode.org/emoji/charts/full-emoji-modifiers.html), an emoji modifier sequence is a sequence of two code points. And from [The Unicode® Standard, Version 14.0 – Core Specification](https://www.unicode.org/versions/Unicode14.0.0/UnicodeStandard-14.0.pdf), page 884, "Emoji Modifiers. The emoji modifiers U+1F3FB..U+1F3FF designate five different skin tones based on the Fitzpatrick scale. These may be displayed in isolation as color or halftone swatches, or they may form a ligature with a preceding emoji character representing a person or body part in order to specify a particular appearance for that character." Using Emacs, Epsilon, and VSCode, I've verified using the emoji sequences in [this table in wiki](https://unicode.org/emoji/charts/full-emoji-modifiers.html) that the modifier sequences in the table are actually two code points, not one. But, "." does not seem correct because it could be not just any preceding emoji, but a "body part". I still do not know where this grammar came about.

Reply:
grapheme.g4 implements a grapheme cluster breaker. It segments the Unicode code point stream into graphemes. See [Table 1c](https://unicode.org/reports/tr29/#Regex_Definitions) of [Unicode Standard Annex #29](https://unicode.org/reports/tr29/). In addition, the grammar was extended to include Emoji's, which are defined by regular expressions in [Section 1.4.9 EBNF and Regex](https://www.unicode.org/reports/tr51/#EBNF_and_Regex) of [Unicode® Technical Standard #51 UNICODE EMOJI](https://www.unicode.org/reports/tr51/). All this syntax is old-style Java regular expression naming, and it should all be rewritten. I will do with this PR.